lib/ide/ide/src/view/ui_component.rs

```diff
@@ -0,0 +1,18 @@
+//! This module contains the base definition of UiComponent.
+
+use nalgebra::Vector2;
+
+/// A trait defining common interfaces of UI components.
+pub trait UiComponent {
+    /// Sets dimensions.
```

Comment:
There's naming mismatch — how is `size` different from `dimensions`? We can pick either one (I'd default to `size` but its not relevant) but let's try keeping it consistent.